controllers/etcd_controller.go

```diff
@@ -262,10 +275,13 @@ func (r *EtcdReconciler) delete(ctx context.Context, etcd *druidv1alpha1.Etcd) (
 
 	if sets.NewString(etcd.Finalizers...).Has(FinalizerName) {
 		logger.Infof("Removing finalizer (%s) from etcd %s", FinalizerName, etcd.GetName())
-		finalizers := sets.NewString(etcd.Finalizers...)
+		// Deep copy of etcd resource required here to patch the object. Update call results in
+		// StorageError. See also: https://github.com/kubernetes/kubernetes/issues/71139
+		etcdCopy := etcd.DeepCopy()
+		finalizers := sets.NewString(etcdCopy.Finalizers...)
 		finalizers.Delete(FinalizerName)
-		etcd.Finalizers = finalizers.UnsortedList()
-		if err := r.Update(ctx, etcd); client.IgnoreNotFound(err) != nil {
+		etcdCopy.Finalizers = finalizers.UnsortedList()
+		if err := r.Patch(ctx, etcdCopy, client.MergeFrom(etcd)); client.IgnoreNotFound(err) != nil {
```

Comment:
Why not an an `Update` here?

Reply:
As mentioned in the comment, for a resource with deletiontimestamp set, in place patch is fine. Update results in error as it does a get first